---
title: "Deseq2"
output: html_document
date: "2022-09-09"
---
Normalized Counts file is from the results from Lara and Felipe "all" folder 

```{r message= FALSE, warning= FALSE, echo=FALSE}
library(tximport)
library(GenomicFeatures)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(DESeq2)
library(ggplot2)
library(tibble)
```

this is different than the normal because we have a normalized counts file from 
the DESeq2 analysis done previously 

This is for setting up the meta data and DESeq2 object so that you can do your analysis 

The only meta data I am adding here is time point 
```{r message= FALSE, warning= FALSE, echo=FALSE}
sample_meta_data <- read.csv(file= "~/Documents/Mythreye_Lab/anoikis/Mehri_Resha/anoikis/meta_data_anoikis.csv") #this is a DIFFERENT meta data file. Example in github. Include any information in this that you want to compare between samples
sampleCondition <- sample_meta_data$Time_Point #this one you can change to whatever column you want in your meta data for comparisons 
sampleTable <- data.frame(sampleName = sample_meta_data$Sample,
                          fileName = sample_meta_data$Sample,
                          condition = sampleCondition)
sampleTable$condition <- factor(sampleTable$condition)
```

```{r}
data <- read.csv("~/Documents/Mythreye_Lab/anoikis/Mehri_Resha/anoikis/raw_counts_edited.csv")
gene <- as.data.frame(data$GeneID)
countDataMatrix <- as.matrix(data[ ,-1])
rownames(countDataMatrix) <- data[,1]
colData <- as.matrix(sample_meta_data$Sample)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeqDataSetFromMatrix(countData= countDataMatrix, colData= sampleTable, design= ~ condition, tidy= FALSE)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeq(dds)
```

This is going to tell you information about the results from running DESeq2 
```{r}
res <- results(dds)
res
```

This is to save the results so that you can look at them later/ comparisons for analysis 
```{r}
write.csv(res, "~/Desktop/results_anoikis.csv")
```


You can set results to whatever you want to compare from the dds object; see documentation for more if you do not like what it auto selects/ you want more than what this did first 
Here is an example template. There's a # in front of it so that it is not running 
```{r}
res_2 <- results(dds, contrast=c("condition","1","7"))

write.csv(res_2,"~/Desktop/P1_vs_P7.csv")
```

```{r}
res_4 <- results(dds, contrast=c("condition","6.6","0"))

write.csv(res_4,"~/Desktop/P6_vs_P0.csv")
```


This is all for the comparison of 0 vs 7 

```{r echo=FALSE}
resOrdered <- res[order(res$pvalue),]
```

Here is another portion to see what all is in the DESeq2 object 
```{r}
summary(res)
```

```{r}
res05 <- results(dds, alpha=0.05)
summary(res05)
```

Now you can start going through the plot functions and seeing what the data actually looks like 
```{r echo=FALSE}
plotMA(res, ylim=c(-2,2))
```

```{r echo=FALSE}
plotMA(res05, ylim=c(-2,2))
```

```{r}
vsd <- vst(dds, blind=FALSE, nsub= 5000)
```

```{r}
rld <- rlog(dds, blind=FALSE)
```

```{r}
ntd <- normTransform(dds)
```

```{r}
library("vsn")
meanSdPlot(assay(ntd))
```

```{r}
meanSdPlot(assay(vsd))
```

```{r}
sampleDists <- dist(t(assay(vsd)))
```

```{r}
library("RColorBrewer")
library("pheatmap")
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$condition)
colnames(sampleDistMatrix) <- NULL 
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors, annRow= vsd$condition)
```


```{r}
legend_title <- "Passage"
plotPCA(vsd, intgroup=c("condition")) + labs(title= "Principle Component Analysis of the 6 Passages based on RNAseq") + theme(text = element_text(size = 10))+ labs(fill = "Passage") + theme(panel.background = element_blank(), panel.grid.major= element_line()) + theme(axis.line = element_line(colour = "black")) 
```

```{r}
plotDispEsts(dds)
```
```{r}
par(mfrow=c(2,2),mar=c(2,2,1,1))
ylim <- c(-2.5,2.5)
resGA <- results(dds, lfcThreshold=1.5, altHypothesis="greaterAbs")
resLA <- results(dds, lfcThreshold=1.5, altHypothesis="lessAbs")
resG <- results(dds, lfcThreshold=1.5, altHypothesis="greater")
resL <- results(dds, lfcThreshold=1.5, altHypothesis="less")
drawLines <- function() abline(h=c(-1.5,1.5),col="dodgerblue",lwd=2)
plotMA(resGA, ylim=ylim); drawLines()
plotMA(resLA, ylim=ylim); drawLines()
plotMA(resG, ylim=ylim); drawLines()
plotMA(resL, ylim=ylim); drawLines()
```
```{r}
resLFC1.5 <- results(dds, lfcThreshold=1.5)
x <-table(resLFC1.5$padj < 0.05)
```

```{r}
head(x)
```
```{r}
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))

# Add colored points: blue if padj<0.01, red if log2FC>1 and padj<0.05)
with(subset(res, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(res, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```


```{r}
write.csv(resLFC1.5 ,"~/Desktop/anoikis/resLFC1_5.csv")
```

```{r}
library("ggbeeswarm")
geneCounts <- plotCounts(dds, gene = "ENSG00000163735.7", intgroup = c("condition"),
                         returnData = TRUE)
ggplot(geneCounts, aes(x = condition, y = count, color = condition)) +
  scale_y_log10() +  geom_beeswarm(cex = 3) 
```

```{r}
ggplot(geneCounts, aes(x = condition, y = count, color = condition, group = "condition")) +
  scale_y_log10() + geom_point(size = 3) + geom_line()
```
Comparison Group: 0 vs 1 
```{r}
res_2 <- results(dds, contrast=c("condition","0","1"))
```

```{r echo=FALSE}
resOrdered <- res_2[order(res_2$pvalue),]
```
 
```{r}
summary(res_2)
```

```{r echo=FALSE}
plotMA(res_2, ylim=c(-2,2))
```

```{r}
resLFC1.5 <- results(dds, lfcThreshold=1.5, contrast=c("condition","0","1") )
x <-table(resLFC1.5$padj < 0.05)
```

```{r}
head(x)
```

```{r}
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res_2, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))

# Add colored points: blue if padj<0.01, red if log2FC>1 and padj<0.05)
with(subset(res_2, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(res_2, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```

```{r}
write.csv(resLFC1.5 ,"~/Desktop/anoikis/resLFC1_5_0vs1.csv")
```

Comparison Group: 0 vs 1 
```{r}
res_3 <- results(dds, contrast=c("condition","0","4"))
```

```{r echo=FALSE}
resOrdered <- res_3[order(res_3$pvalue),]
```
 
```{r}
summary(res_3)
```

```{r echo=FALSE}
plotMA(res_3, ylim=c(-2,2))
```


```{r}
resLFC1.5 <- results(dds, lfcThreshold=1.5, contrast=c("condition","0","4"))
x <-table(resLFC1.5$padj < 0.05)
```

```{r}
head(x)
```

```{r}
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res_3, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))

# Add colored points: blue if padj<0.01, red if log2FC>1 and padj<0.05)
with(subset(res, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(res, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```

```{r}
write.csv(resLFC1.5 ,"~/Desktop/anoikis/resLFC1_5_0vs4.csv")
```

#Aurora Kinase A 
```{r}
counts <- counts(dds['ENSG00000087586.18',], normalized = TRUE)
col = c("0.0"= "#481567FF", "1.0"= "#2D708EFF", "3.3"= "#29AF7FFF", "4.0"= "#E3E41&FE", "6.6"= "#EDE725FF", "7.0"= "#287C8EFF")
m <- list(counts = as.numeric(counts), group = as.character(sample_meta_data$Time_Point))
m <- as_tibble(m)
q <- ggplot(m, aes(group, counts)) + geom_boxplot(aes(fill= group))
q <- q + labs(y = "Normalized Counts", title = "Expression of AURKA") + theme(legend.position= "none") + labs(x= "Passage") + theme(text = element_text(size = 20)) + theme(panel.background = element_blank(), panel.grid.major= element_line()) + theme(axis.line = element_line(colour = "black"))
q
```
```{r}
counts <- counts(dds['ENSG00000178999.13',], normalized = TRUE)
col = c("0.0"= "#481567FF", "1.0"= "#2D708EFF", "3.3"= "#29AF7FFF", "4.0"= "#E3E41&FE", "6.6"= "#EDE725FF", "7.0"= "#287C8EFF")
m <- list(counts = as.numeric(counts), group = as.character(sample_meta_data$Time_Point))
m <- as_tibble(m)
q <- ggplot(m, aes(group, counts)) + geom_boxplot(aes(fill= group))
q <- q + labs(y = "Normalized Counts", title = "Expression of AURKB") + theme(legend.position= "none") + labs(x= "Passage") + theme(text = element_text(size = 20)) + theme(panel.background = element_blank(), panel.grid.major= element_line()) + theme(axis.line = element_line(colour = "black"))
q
```
```{r}
counts <- counts(dds['ENSG00000141510.18',], normalized = TRUE)
col = c("0.0"= "#481567FF", "1.0"= "#2D708EFF", "3.3"= "#29AF7FFF", "4.0"= "#E3E41&FE", "6.6"= "#EDE725FF", "7.0"= "#287C8EFF")
m <- list(counts = as.numeric(counts), group = as.character(sample_meta_data$Time_Point))
m <- as_tibble(m)
q <- ggplot(m, aes(group, counts)) + geom_boxplot(aes(fill= group))
q <- q + labs(y = "Normalized Counts", title = "Expression of p53") + theme(legend.position= "none") + labs(x= "Passage") + theme(text = element_text(size = 20)) + theme(panel.background = element_blank(), panel.grid.major= element_line()) + theme(axis.line = element_line(colour = "black"))
q
```
```{r}
counts <- counts(dds['ENSG00000105329.11',], normalized = TRUE)
col = c("0.0"= "#481567FF", "1.0"= "#2D708EFF", "3.3"= "#29AF7FFF", "4.0"= "#E3E41&FE", "6.6"= "#EDE725FF", "7.0"= "#287C8EFF")
m <- list(counts = as.numeric(counts), group = as.character(sample_meta_data$Time_Point))
m <- as_tibble(m)
q <- ggplot(m, aes(group, counts)) + geom_boxplot(aes(fill= group))
q <- q + labs(y = "Normalized Counts", title = "Expression of TGFB1") + theme(legend.position= "none") + labs(x= "Passage") + theme(text = element_text(size = 20)) + theme(panel.background = element_blank(), panel.grid.major= element_line()) + theme(axis.line = element_line(colour = "black"))
q
```
```{r}
counts <- counts(dds['ENSG00000049768.16',], normalized = TRUE)
col = c("0.0"= "#481567FF", "1.0"= "#2D708EFF", "3.3"= "#29AF7FFF", "4.0"= "#E3E41&FE", "6.6"= "#EDE725FF", "7.0"= "#287C8EFF")
m <- list(counts = as.numeric(counts), group = as.character(sample_meta_data$Time_Point))
m <- as_tibble(m)
q <- ggplot(m, aes(group, counts)) + geom_boxplot(aes(fill= group))
q <- q + labs(y = "Normalized Counts", title = "Expression of FOXP3") + theme(legend.position= "none") + labs(x= "Passage") + theme(text = element_text(size = 20)) + theme(panel.background = element_blank(), panel.grid.major= element_line()) + theme(axis.line = element_line(colour = "black"))
q
```
```{r}
counts <- counts(dds['ENSG00000136997.21',], normalized = TRUE)
col = c("0.0"= "#481567FF", "1.0"= "#2D708EFF", "3.3"= "#29AF7FFF", "4.0"= "#E3E41&FE", "6.6"= "#EDE725FF", "7.0"= "#287C8EFF")
m <- list(counts = as.numeric(counts), group = as.character(sample_meta_data$Time_Point))
m <- as_tibble(m)
q <- ggplot(m, aes(group, counts)) + geom_boxplot(aes(fill= group))
q <- q + labs(y = "Normalized Counts", title = "Expression of MYC") + theme(legend.position= "none") + labs(x= "Passage") + theme(text = element_text(size = 20)) + theme(panel.background = element_blank(), panel.grid.major= element_line()) + theme(axis.line = element_line(colour = "black"))
q
```
