---
title: "Deseq2"
output: html_document
date: "2022-09-09"
---
Normalized Counts file is from the results from Lara and Felipe "all" folder 

```{r message= FALSE, warning= FALSE, echo=FALSE}
library(tximport)
library(GenomicFeatures)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(DESeq2)
```

this is different than the normal because we have a normalized counts file from 
the DESeq2 analysis done previously 

This is for setting up the meta data and DESeq2 object so that you can do your analysis 

The only meta data I am adding here is time point 
```{r message= FALSE, warning= FALSE, echo=FALSE}
sample_meta_data <- read.csv(file= "~/Desktop/anoikis/meta_data_anoikis.csv") #this is a DIFFERENT meta data file. Example in github. Include any information in this that you want to compare between samples
sampleCondition <- sample_meta_data$Time_Point #this one you can change to whatever column you want in your meta data for comparisons 
sampleTable <- data.frame(sampleName = sample_meta_data$Sample,
                          fileName = sample_meta_data$Sample,
                          condition = sampleCondition)
sampleTable$condition <- factor(sampleTable$condition)
```

```{r}
data <- read.csv("~/Desktop/anoikis/raw_counts_edited.csv")
gene <- as.data.frame(data$GeneID)
countDataMatrix <- as.matrix(data[ ,-1])
rownames(countDataMatrix) <- data[,1]
colData <- as.matrix(sample_meta_data$Sample)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeqDataSetFromMatrix(countData= countDataMatrix, colData= sampleTable, design= ~ condition, tidy= FALSE)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeq(dds)
```

This is going to tell you information about the results from running DESeq2 
```{r}
res <- results(dds)
res
```

This is to save the results so that you can look at them later/ comparisons for analysis 
```{r}
write.csv(res, "~/Desktop/anoikis/results_anoikis.csv")
```


You can set results to whatever you want to compare from the dds object; see documentation for more if you do not like what it auto selects/ you want more than what this did first 
Here is an example template. There's a # in front of it so that it is not running 
```{r}
#res_2 <- results(dds, contrast=c("condition","SAY WHICH ONE","AND THE OTHER"))
```


This is all for the comparison of 0 vs 7 

```{r echo=FALSE}
resOrdered <- res[order(res$pvalue),]
```

Here is another portion to see what all is in the DESeq2 object 
```{r}
summary(res)
```

```{r}
res05 <- results(dds, alpha=0.05)
summary(res05)
```

Now you can start going through the plot functions and seeing what the data actually looks like 
```{r echo=FALSE}
plotMA(res, ylim=c(-2,2))
```

```{r echo=FALSE}
plotMA(res05, ylim=c(-2,2))
```

```{r}
vsd <- vst(dds, blind=FALSE)
```

```{r}
rld <- rlog(dds, blind=FALSE)
```

```{r}
ntd <- normTransform(dds)
```

```{r}
library("vsn")
meanSdPlot(assay(ntd))
```

```{r}
meanSdPlot(assay(vsd))
```

```{r}
sampleDists <- dist(t(assay(vsd)))
```

```{r}
library("RColorBrewer")
library("pheatmap")
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$condition)
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors, annRow= vsd$condition)
```
```{r}
legend_title <- "Passage"
plotPCA(vsd, intgroup=c("condition")) + labs(title= "Principle Component Analysis of the 6 Passages based on RNAseq") + theme(text = element_text(size = 10))+ labs(fill = "Passage") + theme(panel.background = element_blank(), panel.grid.major= element_line()) + theme(axis.line = element_line(colour = "black")) 
```

```{r}
plotDispEsts(dds)
```
```{r}
par(mfrow=c(2,2),mar=c(2,2,1,1))
ylim <- c(-2.5,2.5)
resGA <- results(dds, lfcThreshold=1.5, altHypothesis="greaterAbs")
resLA <- results(dds, lfcThreshold=1.5, altHypothesis="lessAbs")
resG <- results(dds, lfcThreshold=1.5, altHypothesis="greater")
resL <- results(dds, lfcThreshold=1.5, altHypothesis="less")
drawLines <- function() abline(h=c(-1.5,1.5),col="dodgerblue",lwd=2)
plotMA(resGA, ylim=ylim); drawLines()
plotMA(resLA, ylim=ylim); drawLines()
plotMA(resG, ylim=ylim); drawLines()
plotMA(resL, ylim=ylim); drawLines()
```
```{r}
resLFC1.5 <- results(dds, lfcThreshold=1.5)
x <-table(resLFC1.5$padj < 0.05)
```

```{r}
head(x)
```
```{r}
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))

# Add colored points: blue if padj<0.01, red if log2FC>1 and padj<0.05)
with(subset(res, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(res, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```


```{r}
write.csv(resLFC1.5 ,"~/Desktop/anoikis/resLFC1_5.csv")
```

```{r}
library("ggbeeswarm")
geneCounts <- plotCounts(dds, gene = "ENSG00000163735.7", intgroup = c("condition"),
                         returnData = TRUE)
ggplot(geneCounts, aes(x = condition, y = count, color = condition)) +
  scale_y_log10() +  geom_beeswarm(cex = 3) 
```

```{r}
ggplot(geneCounts, aes(x = condition, y = count, color = condition, group = "condition")) +
  scale_y_log10() + geom_point(size = 3) + geom_line()
```
Comparison Group: 0 vs 1 
```{r}
res_2 <- results(dds, contrast=c("condition","0","1"))
```

```{r echo=FALSE}
resOrdered <- res_2[order(res_2$pvalue),]
```
 
```{r}
summary(res_2)
```

```{r echo=FALSE}
plotMA(res_2, ylim=c(-2,2))
```

```{r}
resLFC1.5 <- results(dds, lfcThreshold=1.5, contrast=c("condition","0","1") )
x <-table(resLFC1.5$padj < 0.05)
```

```{r}
head(x)
```

```{r}
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res_2, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))

# Add colored points: blue if padj<0.01, red if log2FC>1 and padj<0.05)
with(subset(res_2, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(res_2, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```

```{r}
write.csv(resLFC1.5 ,"~/Desktop/anoikis/resLFC1_5_0vs1.csv")
```

Comparison Group: 0 vs 1 
```{r}
res_3 <- results(dds, contrast=c("condition","0","4"))
```

```{r echo=FALSE}
resOrdered <- res_3[order(res_3$pvalue),]
```
 
```{r}
summary(res_3)
```

```{r echo=FALSE}
plotMA(res_3, ylim=c(-2,2))
```


```{r}
resLFC1.5 <- results(dds, lfcThreshold=1.5, contrast=c("condition","0","4"))
x <-table(resLFC1.5$padj < 0.05)
```

```{r}
head(x)
```

```{r}
#reset par
par(mfrow=c(1,1))
# Make a basic volcano plot
with(res_3, plot(log2FoldChange, -log10(pvalue), pch=20, main="Volcano plot", xlim=c(-3,3)))

# Add colored points: blue if padj<0.01, red if log2FC>1 and padj<0.05)
with(subset(res, padj<.01 ), points(log2FoldChange, -log10(pvalue), pch=20, col="blue"))
with(subset(res, padj<.01 & abs(log2FoldChange)>2), points(log2FoldChange, -log10(pvalue), pch=20, col="red"))
```

```{r}
write.csv(resLFC1.5 ,"~/Desktop/anoikis/resLFC1_5_0vs4.csv")
```

#Aurora Kinase A 
```{r}
counts <- counts(dds['ENSG00000087586.18',], normalized = TRUE)
col = c("0.0"= "#481567FF", "1.0"= "#2D708EFF", "3.3"= "#29AF7FFF", "4.0"= "#E3E41&FE", "6.6"= "#EDE725FF", "7.0"= "#287C8EFF")
m <- list(counts = as.numeric(counts), group = as.character(sample_meta_data$Time_Point))
m <- as_tibble(m)
q <- ggplot(m, aes(group, counts)) + geom_boxplot(aes(fill= group))
q <- q + labs(y = "Normalized Counts", title = "Expression of AURKA") + theme(legend.position= "none") + labs(x= "Passage") + theme(text = element_text(size = 20)) + theme(panel.background = element_blank(), panel.grid.major= element_line()) + theme(axis.line = element_line(colour = "black"))
q
```
```{r}
counts <- counts(dds['ENSG00000178999.13',], normalized = TRUE)
col = c("0.0"= "#481567FF", "1.0"= "#2D708EFF", "3.3"= "#29AF7FFF", "4.0"= "#E3E41&FE", "6.6"= "#EDE725FF", "7.0"= "#287C8EFF")
m <- list(counts = as.numeric(counts), group = as.character(sample_meta_data$Time_Point))
m <- as_tibble(m)
q <- ggplot(m, aes(group, counts)) + geom_boxplot(aes(fill= group))
q <- q + labs(y = "Normalized Counts", title = "Expression of AURKB") + theme(legend.position= "none") + labs(x= "Passage") + theme(text = element_text(size = 20)) + theme(panel.background = element_blank(), panel.grid.major= element_line()) + theme(axis.line = element_line(colour = "black"))
q
```
```{r}
counts <- counts(dds['ENSG00000141510.18',], normalized = TRUE)
col = c("0.0"= "#481567FF", "1.0"= "#2D708EFF", "3.3"= "#29AF7FFF", "4.0"= "#E3E41&FE", "6.6"= "#EDE725FF", "7.0"= "#287C8EFF")
m <- list(counts = as.numeric(counts), group = as.character(sample_meta_data$Time_Point))
m <- as_tibble(m)
q <- ggplot(m, aes(group, counts)) + geom_boxplot(aes(fill= group))
q <- q + labs(y = "Normalized Counts", title = "Expression of p53") + theme(legend.position= "none") + labs(x= "Passage") + theme(text = element_text(size = 20)) + theme(panel.background = element_blank(), panel.grid.major= element_line()) + theme(axis.line = element_line(colour = "black"))
q
```
```{r}
counts <- counts(dds['ENSG00000105329.11',], normalized = TRUE)
col = c("0.0"= "#481567FF", "1.0"= "#2D708EFF", "3.3"= "#29AF7FFF", "4.0"= "#E3E41&FE", "6.6"= "#EDE725FF", "7.0"= "#287C8EFF")
m <- list(counts = as.numeric(counts), group = as.character(sample_meta_data$Time_Point))
m <- as_tibble(m)
q <- ggplot(m, aes(group, counts)) + geom_boxplot(aes(fill= group))
q <- q + labs(y = "Normalized Counts", title = "Expression of TGFB1") + theme(legend.position= "none") + labs(x= "Passage") + theme(text = element_text(size = 20)) + theme(panel.background = element_blank(), panel.grid.major= element_line()) + theme(axis.line = element_line(colour = "black"))
q
```
```{r}
counts <- counts(dds['ENSG00000049768.16',], normalized = TRUE)
col = c("0.0"= "#481567FF", "1.0"= "#2D708EFF", "3.3"= "#29AF7FFF", "4.0"= "#E3E41&FE", "6.6"= "#EDE725FF", "7.0"= "#287C8EFF")
m <- list(counts = as.numeric(counts), group = as.character(sample_meta_data$Time_Point))
m <- as_tibble(m)
q <- ggplot(m, aes(group, counts)) + geom_boxplot(aes(fill= group))
q <- q + labs(y = "Normalized Counts", title = "Expression of FOXP3") + theme(legend.position= "none") + labs(x= "Passage") + theme(text = element_text(size = 20)) + theme(panel.background = element_blank(), panel.grid.major= element_line()) + theme(axis.line = element_line(colour = "black"))
q
```
```{r}
counts <- counts(dds['ENSG00000136997.21',], normalized = TRUE)
col = c("0.0"= "#481567FF", "1.0"= "#2D708EFF", "3.3"= "#29AF7FFF", "4.0"= "#E3E41&FE", "6.6"= "#EDE725FF", "7.0"= "#287C8EFF")
m <- list(counts = as.numeric(counts), group = as.character(sample_meta_data$Time_Point))
m <- as_tibble(m)
q <- ggplot(m, aes(group, counts)) + geom_boxplot(aes(fill= group))
q <- q + labs(y = "Normalized Counts", title = "Expression of MYC") + theme(legend.position= "none") + labs(x= "Passage") + theme(text = element_text(size = 20)) + theme(panel.background = element_blank(), panel.grid.major= element_line()) + theme(axis.line = element_line(colour = "black"))
q
```
#Graphs from Lara with comparisons by group
#they defined up and donw regulation (lfc of 1.5 in this option ) and had the passage comparison.
Need a few things for this to work 
1) On each of the files we are comparing, we need a data column that defines up vs down regulated genes
2) the data of each group 
3) ggplot bar graph for these comparisons 
4) Something on the x axis for what is being compared
5) Make the data make more sense compared to the current figures 

Things I want included in the figure 
ADJUSTED BY THE ADJUSTED P VALUE FIRST. NON SIGNIFICANT P VALS WERE DELETED IN MY NEW FILE 
1) up regulated > LFC 1.5 
2) Down regulated < -1.5 
3) No change as between 1.5 and -1.5 
Bar Graph showing all 3 for each comparison group 
Table showing the percentage of genes in each category 
Files for this are labeled as Comparison Group_Gene_Annotation
Column of interest will be labeled as Group
```{r}
P0_vs_P1 <- read.csv("~/Desktop/P1_vs_P0_Gene_Annotation.csv")
P0_vs_P4 <- read.csv("~/Desktop/P4_vs_P0_Gene_Annotation.csv")
P0_vs_P7 <- read.csv("~/Desktop/P7_vs_P0_Gene_Annotation.csv")
```

```{r}
x <- list(P0_vs_P1= P0_vs_P1,P0_vs_P4= P0_vs_P4,P0_vs_P7= P0_vs_P7)
level_order <- c("No Change", "Down Regulated", "Up Regulated")
#for the actual bar graph itself 
library(ggplot2)
ggplot() +
  geom_bar(data = P0_vs_P1, mapping = aes(x= Group, group =Group, fill = Group), position = "dodge") + xlab("P0 vs P1") + ylab("Total Number of Genes")  
```
```{r}
#for the actual bar graph itself 
ggplot() +
  geom_bar(data = P0_vs_P4, mapping = aes(x= Group, group =Group, fill = Group), position = "dodge") + xlab("P0 vs P4") + ylab("Total Number of Genes") 
```
```{r}
#for the actual bar graph itself 
ggplot() +
  geom_bar(data = P0_vs_P7, mapping = aes(x= Group, group =Group, fill = Group), position = "dodge") + xlab("P0 vs P7") + ylab("Total Number of Genes") 
```
I went ahead and manually counted the individual points that are present above so it can be in one graph 
```{r}
library(tidyr)
data <- read.csv("~/Desktop/Groups_by_Gene_Count.csv")
data3 <- data[ -1, ]
```


```{r}
data2 <- data3 %>%
  pivot_longer(cols= P0.vs.P1:P0.vs.P7, names_to = "time_point", values_to= "counts"
)
```


```{r}
#for the actual bar graph itself 
ggplot() +
  geom_col(data = data2, mapping = aes(x= time_point, group= Group, y= counts, fill= Group), position = "dodge") + xlab("Comparison Groups") + ylab("Total Number of Genes") 
```
#The rest of Lara's figure is made with UpSet Figures 
## Need to use ComplexUpSet because the other one is out of date (yay Lara for telling me)
```{r}
library(ggplot2)
library(tidyr)
library(ComplexUpset)
```

#THIS WORKED TO MAKE THE RIGHT DATASET. Hopefully 
```{r}
data <- read.csv("~/Desktop/data_table_regulation_comparison.csv")
data2 <- data %>%
  pivot_longer(cols= c(Group.P0_P1,Group.P0_P4,Group.P0_P7), names_to = "time_point", values_to= c("Group")
)

Group <- data2$time_point
```

Another option 
1. Subset the separate Values from the count files and assign a unique variable to each 
2. Overlaps from ComplexUpset
3. Graph accordingly 

```{r}
P0_vs_P1 <- read.csv("~/Downloads/results/all/comparisons/group_P1_3D_vs_P0_2D_Wald.csv")
P0_vs_P1_Up <- subset(P0_vs_P1[ which(P0_vs_P1$log2FoldChange > 1.5 
& P0_vs_P1$padj < 0.05), ])
P0_vs_P1_Down <- subset(P0_vs_P1[ which(P0_vs_P1$log2FoldChange < -1.5 
& P0_vs_P1$padj < 0.05), ])
```

```{r}
P0_vs_P4 <- read.csv("~/Downloads/results/all/comparisons/group_P4_3D_vs_P0_2D_Wald.csv")
P0_vs_P4_Up <- subset(P0_vs_P4[ which(P0_vs_P4$log2FoldChange > 1.5 
& P0_vs_P4$padj < 0.05), ])
P0_vs_P4_Down <- subset(P0_vs_P4[ which(P0_vs_P4$log2FoldChange < -1.5 
& P0_vs_P4$padj < 0.05), ])
```

```{r}
P0_vs_P7 <- read.csv("~/Downloads/results/all/comparisons/group_P7_3D_vs_P0_2D_Wald.csv")
P0_vs_P7_Up <- subset(P0_vs_P7[ which(P0_vs_P7$log2FoldChange > 1.5 
& P0_vs_P7$padj < 0.05), ])
P0_vs_P7_Down <- subset(P0_vs_P7[ which(P0_vs_P7$log2FoldChange < -1.5 
& P0_vs_P7$padj < 0.05), ])
```

```{r}
data <- as.matrix(x= c("P0_vs_P1 up-regulated" = P0_vs_P1_Up,
                   "P0_vs_P1 down-regulated" = P0_vs_P1_Down,
                   "P0_vs_P4 up-regulated" = P0_vs_P4_Up,
                   "P0_vs_P4 down-regulated" = P0_vs_P4_Down,
                   "P0_vs_P7 up-regulated" = P0_vs_P7_Up,
                   "P0_vs_P7 down-regulated" = P0_vs_P7_Down))

```

```{r}
upset_gene <- upset(data, set_sizes=FALSE,
      c('P0_vs_P1 up-regulated.gene_id', 'P0_vs_P1 down-regulated_.gene_id', 
        'P0_vs_P4 up-regulated.gene_id', 'P0_vs_P4 down-regulated.gene_id', 
        'P0_vs_P7 up-regulated.gene_id', 'P0_vs_P7 down-regulated.gene_id'),
  queries=list(
    upset_query(set='P0_vs_P1 up-regulated.gene_id', fill='red'),
    upset_query(set='P0_vs_P1 down-regulated.gene_id', fill='blue'),
    upset_query(set='P0_vs_P4  up-regulated.gene_id', fill='red'),
    upset_query(set='P0_vs_P4 down-regulated.gene_id', fill='blue'),
    upset_query(set='P0_vs_P7  up-regulated.gene_id', fill='red'),
    upset_query(set='P0_vs_P7 down-regulated.gene_id', fill='blue')
  ),
  base_annotations=list(
    'Intersection size'=(
        intersection_size(
        bar_number_threshold=1,  # show all numbers on top of bars
        width=0.5,   # reduce width of the bars
      )
      # add some space on the top of the bars
      + scale_y_continuous(expand=expansion(mult=c(0, 0.05)))
      + theme(
          # hide grid lines
          panel.grid.major=element_blank(),
          panel.grid.minor=element_blank(),
          # show axis lines
          axis.line=element_line(colour='black')
      )
    )
  ),
  stripes=upset_stripes(
    geom=geom_segment(size=12),  # make the stripes larger
    colors=c('grey95', 'white')
  ),
  # to prevent connectors from getting the colorured
  # use `fill` instead of `color`, together with `shape='circle filled'`
  matrix=intersection_matrix(
      geom=geom_point(
        shape='circle filled',
        size=3,
        stroke=0.45
      )
  ),
  sort_sets='descending',
  sort_intersections='descending'
)
upset_gene
```

