---
title: "RNASeq"
output: html_document
date: "2023-01-06"
---

```{r message= FALSE, warning= FALSE, echo=FALSE}
library(tximport)
library(GenomicFeatures)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(DESeq2)
library(ggplot2)
library(tibble)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
sample_meta_data <- read.csv(file= "~/Documents/Mythreye_Lab/Anoikis_Project/Bulk_RNA/CAOV3/meta_CAOV3.csv") #this is a DIFFERENT meta data file. Example in github. Include any information in this that you want to compare between samples
sampleCondition <- sample_meta_data$Time_Point #this one you can change to whatever column you want in your meta data for comparisons 
sampleTable <- data.frame(sampleName = sample_meta_data$Sample,
                          fileName = sample_meta_data$Sample,
                          condition = sampleCondition)
sampleTable$condition <- factor(sampleTable$condition)
```

```{r}
data <- read.csv("~/RNAseq_Anoikis/CAOV3/Deseq2/feature_counts_unstranded_CAOV3.csv")
gene <- as.data.frame(data$Gene)
countDataMatrix <- as.matrix(data[ ,-1])
rownames(countDataMatrix) <- data[,1]
colData <- as.matrix(sample_meta_data$Sample)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeqDataSetFromMatrix(countData= countDataMatrix, colData= sampleTable, design= ~ condition, tidy= FALSE)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeq(dds)
```

```{r}
counts <- counts(dds[ , ], normalized = TRUE)
write.csv(counts, "~/Desktop/normalized_counts_CAOV3.csv")
```

This is going to tell you information about the results from running DESeq2 
```{r}
res <- results(dds, contrast=c("condition","7","0"))
res
```

This is to save the results so that you can look at them later/ comparisons for analysis 
```{r}
write.csv(res, "~/Desktop/results_7_vs_0_CAOV3.csv")
```


You can set results to whatever you want to compare from the dds object; see documentation for more if you do not like what it auto selects/ you want more than what this did first 
Here is an example template. There's a # in front of it so that it is not running 
```{r}
res_0_vs_1 <- results(dds, contrast=c("condition","1","0"))
write.csv(res_0_vs_1, "~/Desktop/results_1_vs_0_CAOV3.csv")
res_0_vs_6 <- results(dds, contrast=c("condition","6","0"))
write.csv(res_0_vs_6, "~/Desktop/results_6_vs_0_CAOV3.csv")
res_1_vs_7 <- results(dds, contrast=c("condition","7","1"))
write.csv(res_0_vs_6, "~/Desktop/results_7_vs_1_CAOV3.csv")
res_6_7 <- results(dds, contrast=c("condition","7","6"))
write.csv(res_6_7,"~/Desktop/P6_vs_P7_CAOV3.csv")
```


```{r echo=FALSE}
resOrdered <- res[order(res$pvalue),]
```

Here is another portion to see what all is in the DESeq2 object 
```{r}
summary(res)
```

```{r}
res05 <- results(dds, alpha=0.05)
summary(res05)
```

Now you can start going through the plot functions and seeing what the data actually looks like 
```{r echo=FALSE}
plotMA(res, ylim=c(-2,2))
```

```{r echo=FALSE}
plotMA(res05, ylim=c(-2,2))
```

```{r}
vsd <- vst(dds, blind=FALSE)
```

```{r}
rld <- rlog(dds, blind=FALSE)
```

```{r}
ntd <- normTransform(dds)
```

```{r}
library("vsn")
meanSdPlot(assay(ntd))
```

```{r}
meanSdPlot(assay(vsd))
```

```{r}
sampleDists <- dist(t(assay(vsd)))
```

```{r}
library("RColorBrewer")
library("pheatmap")
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$condition)
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors, annRow= vsd$condition)
```

```{r}
plotPCA(vsd, intgroup=c("condition"))
```

```{r}
plotDispEsts(dds)
```




