#P0 (minus the first)
#bam <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P02/Aligned.sortedByCoord.out.bam"
#out <- "data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P02/P02_counts.csv"
#gtf <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/GRCH38/gencode_v37_annotation.gtf"
#library("Rsubread")
#X <- featureCounts(bam,  annot.ext= gtf, isGTFAnnotationFile= TRUE)
#write.csv(X, out)

#bam <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P03/Aligned.sortedByCoord.out.bam"
#out <- "data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P03/P03_counts.csv"
#gtf <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/GRCH38/gencode_v37_annotation.gtf"
#library("Rsubread")
#X <- featureCounts(bam,  annot.ext= gtf, isGTFAnnotationFile= TRUE)
#write.csv(X, out)

#P1 
#bam <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P11/Aligned.sortedByCoord.out.bam"
#out <- "data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P11/P11_counts.csv"
#gtf <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/GRCH38/gencode_v37_annotation.gtf"
#library("Rsubread")
#X <- featureCounts(bam,  annot.ext= gtf, isGTFAnnotationFile= TRUE)
#write.csv(X, out)

#bam <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P12/Aligned.sortedByCoord.out.bam"
#out <- "data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P12/P12_counts.csv"
#gtf <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/GRCH38/gencode_v37_annotation.gtf"
#library("Rsubread")
#X <- featureCounts(bam,  annot.ext= gtf, isGTFAnnotationFile= TRUE)
#write.csv(X, out)

#bam <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P13/Aligned.sortedByCoord.out.bam"
#out <- "data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P13/P13_counts.csv"
#gtf <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/GRCH38/gencode_v37_annotation.gtf"
#library("Rsubread")
#X <- featureCounts(bam,  annot.ext= gtf, isGTFAnnotationFile= TRUE)
#write.csv(X, out)

#P6
#bam <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P61/Aligned.sortedByCoord.out.bam"
#out <- "data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P61/P61_counts.csv"
#gtf <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/GRCH38/gencode_v37_annotation.gtf"
#library("Rsubread")
#X <- featureCounts(bam,  annot.ext= gtf, isGTFAnnotationFile= TRUE)
#write.csv(X, out)

#bam <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P62/Aligned.sortedByCoord.out.bam"
#out <- "data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P62/P62_counts.csv"
#gtf <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/GRCH38/gencode_v37_annotation.gtf"
#library("Rsubread")
#X <- featureCounts(bam,  annot.ext= gtf, isGTFAnnotationFile= TRUE)
#write.csv(X, out)

#bam <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P63/Aligned.sortedByCoord.out.bam"
#out <- "data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P63/P63_counts.csv"
#gtf <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/GRCH38/gencode_v37_annotation.gtf"
#library("Rsubread")
#X <- featureCounts(bam,  annot.ext= gtf, isGTFAnnotationFile= TRUE)
#write.csv(X, out)

#P7
bam <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P71/Aligned.sortedByCoord.out.bam"
out <- "data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P71/P71_counts.csv"
gtf <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/GRCH38/gencode_v37_annotation.gtf"
library("Rsubread")
X <- featureCounts(bam,  annot.ext= gtf, isGTFAnnotationFile= TRUE)
write.csv(X, out)

bam <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P72/Aligned.sortedByCoord.out.bam"
out <- "data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P72/P72_counts.csv"
gtf <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/GRCH38/gencode_v37_annotation.gtf"
library("Rsubread")
X <- featureCounts(bam,  annot.ext= gtf, isGTFAnnotationFile= TRUE)
write.csv(X, out)

bam <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P73/Aligned.sortedByCoord.out.bam"
out <- "data/project/sloss/Mythreye_lab/Sequencing_Data/Anoikis/CAOV3/CAOV3_P0_P1_P6_P7_Bulk_RNAseq_Jan_2023/Trim_Galore_CAOV3/P73/P73_counts.csv"
gtf <- "/data/project/sloss/Mythreye_lab/Sequencing_Data/GRCH38/gencode_v37_annotation.gtf"
library("Rsubread")
X <- featureCounts(bam,  annot.ext= gtf, isGTFAnnotationFile= TRUE)
write.csv(X, out)
