---
title: "Treatment15U"
output: html_document
date: "2023-03-23"
---
15U 
This is to compare the groups of the anoikis project that have been treated with 15U

```{r message= FALSE, warning= FALSE, echo=FALSE}
library(tximport)
library(GenomicFeatures)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(DESeq2)
library(EnhancedVolcano)
library(org.Hs.eg.db)
library(dplyr)
library(tidyr)
```

This is for setting up the meta data and DESeq2 object so that you can do your analysis 
```{r message= FALSE, warning= FALSE, echo=FALSE}
sample_meta_data <- read.csv(file= "C:/Users/empage/Documents/Lab_Stuff/Anoikis/OV90/treatment_15U/meta_15U_updated.csv") 
sampleFiles <- samples$Sample_ID
sampleCondition <- sample_meta_data$Passage_ID
sampleTable <- data.frame(sampleName = sampleFiles,
                          fileName = sampleFiles,
                          condition = sampleCondition)
sampleTable$condition <- factor(sampleTable$condition)
```

```{r}
data <- read.csv("~/RNAseq_Anoikis/OV90/RNAseq/15U/normalization_both_groups/raw_counts_feature_counts.csv")
gene <- as.data.frame(data$GeneID)
countDataMatrix <- as.matrix(data[ ,-1])
rownames(countDataMatrix) <- data[,1]
colData <- as.matrix(sample_meta_data$Sample)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeqDataSetFromMatrix(countData= countDataMatrix, colData= sampleTable, design= ~ condition, tidy= FALSE)
```

```{r message= FALSE, warning= FALSE, echo=FALSE}
dds <- DESeq(dds)
```


```{r}
counts <- counts(dds[ , ], normalized = TRUE)
write.csv(counts, "~/RNAseq_Anoikis/OV90/RNAseq/15U/normalized_counts_15U.csv")
```

  
This is going to tell you information about the results from running DESeq2 
```{r}
res <- results(dds, contrast= c("condition", "P1_15U","P0_15U"))
write.csv(res_3, "~/RNAseq_Anoikis/OV90/RNAseq/15U/normalization_both_groups/res/P0_P1_15U.csv")
```

```{r echo=FALSE}
plotMA(res, ylim=c(-2,2))
```
You can set results to whatever you want to compare from the dds object; see documentation for more if you do not like what it auto selects/ you want more than what this did first 
Here is an example template. There's a # in front of it so that it is not running 
```{r}
res_2 <- results(dds, contrast=c("condition","P1.1_15U", "P0_15U"))
write.csv(res_2, "~/RNAseq_Anoikis/OV90/RNAseq/15U/normalization_both_groups/res/P0_P11_15U.csv")
```

```{r echo=FALSE}
plotMA(res_2, ylim=c(-2,2))
```

```{r}
res_3 <- results(dds, contrast=c("condition","P1.1_15U","P1_15U"))
write.csv(res_3, "~/RNAseq_Anoikis/OV90/RNAseq/15U/normalization_both_groups/res/P1_P11_15U.csv")
```

```{r echo=FALSE}
plotMA(res_3, ylim=c(-2,2))
```

```{r}
res_4 <- results(dds, contrast=c("condition","P1.1_DMSO","P1_DMSO"))
summary(res_4)
write.csv(res_4, "~/RNAseq_Anoikis/OV90/RNAseq/15U/normalization_both_groups/res/P1_P11_DMSO.csv")
```
```{r echo=FALSE}
plotMA(res_4, ylim=c(-2,2))
```

```{r}
res_5 <- results(dds, contrast=c("condition", "P1.1_DMSO", "P0_DMSO"))
summary(res_5)
write.csv(res_5, "~/RNAseq_Anoikis/OV90/RNAseq/15U/normalization_both_groups/res/P0_P11_DMSO.csv")
```
```{r echo=FALSE}
plotMA(res_5, ylim=c(-2,2))
```

```{r}
res_6 <- results(dds, contrast=c("condition", "P1_DMSO", "P0_DMSO"))
summary(res_6)
write.csv(res_6, "~/RNAseq_Anoikis/OV90/RNAseq/15U/normalization_both_groups/res/P0_P1_DMSO.csv")
```

```{r echo=FALSE}
plotMA(res_6, ylim=c(-2,2))
```

```{r}
res_7 <- results(dds, contrast=c("condition","P0_15U","P0_DMSO"))
summary(res_7)
write.csv(res_7, "~/RNAseq_Anoikis/OV90/RNAseq/15U/normalization_both_groups/res/P0_15U_DMSO.csv")
```

```{r echo=FALSE}
plotMA(res_7, ylim=c(-2,2))
```

```{r}
res_8 <- results(dds, contrast=c("condition","P1_15U","P1_DMSO"))
summary(res_8)
write.csv(res_8, "~/RNAseq_Anoikis/OV90/RNAseq/15U/normalization_both_groups/res/P1_15U_DMSO.csv")
```

```{r echo=FALSE}
plotMA(res_8, ylim=c(-2,2))
```

```{r}
res_9 <- results(dds, contrast=c("condition","P1.1_15U","P1.1_DMSO"))
summary(res_9)
write.csv(res_9, "~/RNAseq_Anoikis/OV90/RNAseq/15U/normalization_both_groups/res/P11_15U_DMSO.csv")
```

```{r}
res_10 <- results(dds, contrast=c("condition","P1_15U","P0_DMSO"))
summary(res_10)
write.csv(res_10, "~/RNAseq_Anoikis/OV90/RNAseq/15U/normalization_both_groups/res/P1_15U_DMSO.csv")
```

```{r echo=FALSE}
plotMA(res_9, ylim=c(-2,2))
```
for the object as a whole  
```{r}
vsd <- vst(dds, blind=FALSE)
```

```{r}
rld <- rlog(dds, blind=FALSE)
```

```{r}
ntd <- normTransform(dds)
```

```{r}
plotPCA(vsd, intgroup=c("condition"))
```

```{r}
plotDispEsts(dds)
```

```{r}
vsd <- vst(dds, blind=FALSE)
```

```{r}
rld <- rlog(dds, blind=FALSE)
```

```{r}
ntd <- normTransform(dds)
```

```{r}
library("vsn")
meanSdPlot(assay(ntd))
```

```{r}
meanSdPlot(assay(vsd))
```

```{r}
sampleDists <- dist(t(assay(vsd)))
```

```{r}
library("RColorBrewer")
library("pheatmap")
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$condition)
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors, annRow= vsd$condition)
```

